src/main/java/ev3dev/utils/DataChannelRereader.java

```diff
@@ -50,15 +50,14 @@ public DataChannelRereader(String pathString) {
      */
     public String readString() {
         try {
-            int n;
-            do {
-                byteBuffer.clear();
-                channel.position(0);
-                n = channel.read(byteBuffer);
-                if (n == -1) {
-                    throw new IOException("Premature end of file ");
-                }
-            } while (n <= 0);
+            byteBuffer.clear();
+            channel.position(0);
+            int n = channel.read(byteBuffer);
+            if (n == -1) {
```

Comment:
I think this branch should handle the `n == 0` case as well. I'm not sure when would this happen, but if it happens then the code below will break (`bytes[-1]` is invalid).

Reply:
It is not a bad idea.
I will merge this version and later we will refine with that idea

Reply:
This was already added: https://github.com/ev3dev-lang-java/ev3dev-lang-java/pull/778/commits/c1b78979bd5d9dd4eb638b855153f2ec42fcb868 (later commit in this PR)